src/address.c

```diff
@@ -40,27 +39,27 @@ bool address_from_pubkey(const uint8_t *public_key, char *out, size_t out_len) {
         return false;
     }
 
-    uint8_t core[ADDRESS_LEN] = {0};
+    uint8_t core[ADDRESS_LEN];
     core[0] = 0;  // user byte
-    memmove(core + 1, hash, ADDRESS_LEN - 1);
+    memcpy(core + 1, hash, ADDRESS_LEN - 1);
 
     return address_encode(core, out, out_len);
 }
 
 bool address_encode(const uint8_t *prog, char *out, size_t out_len) {
-    char bech32[73 + 1] = {0};
-
+    // Ensure the output buffer is large enough
     if (41 > out_len) {
         return false;
     }
 
+    char bech32[73];
+
     if (bech32_addr_encode(bech32, "z", prog, ADDRESS_LEN) != 1) {
```

Comment:
can you add a comment about what this is checking

Reply:
added comments to both functions and removed the explicit zero initialization of the hash variable.

Reply:
thanks looking again